core/utils.class.inc.php

```diff
@@ -713,29 +718,45 @@ public static function CheckModuleInstallation(string $sModuleId, bool $bRequire
 				static::$sLastInstallDate = $aDatamodel['fields']['installed'];
 			}
 
-			$aResults = $oClient->Get('ModuleInstallation', ['name' => $sName, 'installed' => static::$sLastInstallDate], 'name,version', 1);
-			if ($aResults['code'] != 0 || empty($aResults['objects'])) {
-				throw new Exception($aResults['message'], $aResults['code']);
+			if (!isset(static::$aModuleVersions[$sName])) {
+				$aResults = $oClient->Get('ModuleInstallation', ['name' => $sName, 'installed' => static::$sLastInstallDate], 'version', 1);
+				if ($aResults['code'] != 0 || empty($aResults['objects'])) {
+					throw new Exception($aResults['message'], $aResults['code']);
+				}
+				$aObject = current($aResults['objects']);
+				static::$aModuleVersions[$sName] = $aObject['fields']['version'];
 			}
-			$aObject = current($aResults['objects']);
-			$sCurrentVersion = $aObject['fields']['version'];
 
-			if (isset($sExpectedVersion) && !version_compare($sCurrentVersion, $sExpectedVersion, $sOperator)) {
-				throw new Exception(sprintf('Version mismatch (%s %s %s)', $sCurrentVersion, $sOperator, $sExpectedVersion));
+			if (isset($sExpectedVersion) && !version_compare(static::$aModuleVersions[$sName], $sExpectedVersion, $sOperator)) {
+				throw new Exception(sprintf('Version mismatch (%s %s %s)', static::$aModuleVersions[$sName], $sOperator, $sExpectedVersion));
 			}
 
-			Utils::Log(LOG_DEBUG, sprintf('iTop module %s version %s is installed.', $aObject['fields']['name'], $sCurrentVersion));
+			Utils::Log(LOG_DEBUG, sprintf('iTop module %s version %s is installed.', $sName, static::$aModuleVersions[$sName]));
 		} catch (Exception $e) {
 			$sMessage = sprintf('%s iTop module %s is considered as not installed due to: %s', $bRequired ? 'Required' : 'Optional', $sName, $e->getMessage());
 			if ($bRequired) {
-				throw new Exception($sMessage, 0, $e);
+				throw new Exception($sMessage, previous: $e);
```

Comment:
not compatible PHP 7.x

------ ---------------------------------------------------------- 
  Line   core/utils.class.inc.php                                  
 ------ ---------------------------------------------------------- 
  738    Named arguments are supported only on PHP 8.0 and later.  
 ------ ----------------------------------------------------------

Reply:
Good catch, I had my IDE settings wrong. All fixed now.